build-and-test.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -e 	# Exit immediately upon failure
+set -o pipefail  # Carry failures over pipes
+
+repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+docker_repo="microsoft/dotnet-nightly"
+
+function build_dockerfiles {
+    for dockerfile_dir in $( egrep -v 'nanoserver|windowsservercore' <<< "${1}" | sort ); do
+        tag="${docker_repo}:$( sed -e 's/.\///' -e 's/debian\///' -e 's/debian/sdk/' -e 's/\//-/g' <<< "${dockerfile_dir}" )"
+        echo "----- Building ${tag} -----"
+        docker build -t "${tag}" "${dockerfile_dir}"
+    done
+}
+
+pushd "${repo_root}" > /dev/null
+
+build_dockerfiles "$( find . -path './.*' -prune -o -name 'Dockerfile' -a -path '*/*-deps/*' -print0 | xargs -0 -n1 dirname )"
```

Comment:
I wish there was a better way to express the image dependencies.  I guess we can refactor as we go :)

---
In reply to: [76364173](https://github.com/dotnet/dotnet-docker-nightly/pull/35#discussion_r76364173) [](ancestors = 76364173)